### PR TITLE
Update benches that are affected by receipts

### PR DESCRIPTION
**Pull request by Dentosal**
Closes https://github.com/FuelLabs/fuel-core/issues/1414. Also does some upgrades for latest `fuel-vm` version. Targets https://github.com/FuelLabs/fuel-vm/pull/625 currently, needs new `fuel-vm` release after that is merged.

This also mostly reverts https://github.com/FuelLabs/fuel-core/pull/1457 which was not working correctly, especially with regards to `call` instruction.

### CODE REVIEW

**Review by xgreenx — COMMENTED** [review]
Could you post a difference in results please?=)

**Comment by Dentosal** [comment]
At 5f5ea52d

```
add: 1
addi: 1
aloc: 1
and: 1
andi: 1
bal: 17
bhei: 1
bhsh: 1
burn: 9333
call:
  base: 3344
  dep_per_unit: 29
cb: 1
ccp:
  base: 24
  dep_per_unit: 159
cfei: 1
cfsi: 1
croo: 23
csiz:
  base: 30
  dep_per_unit: 420
div: 1
divi: 1
eck1: 2016
ecr1: 28162
ed19: 2582
eq: 1
exp: 1
expi: 1
flag: 1
gm: 1
gt: 1
gtf: 1
ji: 1
jmp: 1
jne: 1
jnei: 1
jnzi: 1
k256:
  base: 27
  dep_per_unit: 5
lb: 1
ldc:
  base: 25
  dep_per_unit: 162
log: 364
logd:
  base: 685
  dep_per_unit: 3
lt: 1
lw: 1
mcl:
  base: 1
  dep_per_unit: 1388
mcli:
  base: 1
  dep_per_unit: 1388
mcp:
  base: 2
  dep_per_unit: 1269
mcpi:
  base: 3
  dep_per_unit: 1365
meq:
  base: 1
  dep_per_unit: 1975
mint: 24430
mlog: 1
mod: 1
modi: 1
move: 1
movi: 1
mroo: 3
mul: 1
muli: 1
noop: 1
not: 1
or: 1
ori: 1
ret_contract: 421
ret_script: 381
retd_contract:
  base: 712
  dep_per_unit: 3
retd_script:
  base: 700
  dep_per_unit: 3
rvrt_contract: 402
rvrt_script: 358
s256:
  base: 32
  dep_per_unit: 4
sb: 1
sll: 1
slli: 1
smo:
  base: 19848
  dep_per_unit: 2
srl: 1
srli: 1
srw: 51
srwq:
  base: 93
  dep_per_unit: 1
sub: 1
subi: 1
sw: 1
sww: 168
time: 1
tr: 16415
tro: 9406
wdam: 5
wdcm: 1
wddv: 3
wdmd: 7
wdml: 2
wdmm: 6
wdop: 2
wqam: 7
wqcm: 1
wqdv: 4
wqmd: 11
wqml: 2
wqmm: 7
wqop: 2
xor: 1
xori: 1
```

**Comment by Dentosal** [comment]
At a53fe1db

```
add: 1
addi: 1
aloc: 1
and: 1
andi: 1
bal: 494
bhei: 1
bhsh: 1
burn: 36008
call:
  base: 10917
  dep_per_unit: 9
cb: 1
ccp:
  base: 218
  dep_per_unit: 56
cfei: 1
cfsi: 1
contract_root:
  base: 38
  dep_per_unit: 2
croo: 126
csiz:
  base: 195
  dep_per_unit: 74
div: 1
divi: 1
eck1: 2446
ecr1: 34139
ed19: 3427
eq: 1
exp: 1
expi: 1
flag: 1
gm: 1
gt: 1
gtf: 1
ji: 1
jmp: 1
jne: 1
jnei: 1
jnzi: 1
k256:
  base: 32
  dep_per_unit: 4
lb: 1
ldc:
  base: 221
  dep_per_unit: 57
log: 909
logd:
  base: 3798
  dep_per_unit: 3
lt: 1
lw: 1
mcl:
  base: 1
  dep_per_unit: 555
mcli:
  base: 1
  dep_per_unit: 564
mcp:
  base: 2
  dep_per_unit: 552
mcpi:
  base: 4
  dep_per_unit: 1023
meq:
  base: 2
  dep_per_unit: 1428
mint: 30402
mlog: 1
mod: 1
modi: 1
move: 1
movi: 1
mroo: 3
mul: 1
muli: 1
noop: 1
not: 1
or: 1
ori: 1
ret_contract: 1001
ret_script: 900
retd_contract:
  base: 3819
  dep_per_unit: 2
retd_script:
  base: 3838
  dep_per_unit: 3
rvrt_contract: 1006
rvrt_script: 903
s256:
  base: 38
  dep_per_unit: 3
sb: 1
sll: 1
slli: 1
smo:
  base: 63740
  dep_per_unit: 1
srl: 1
srli: 1
srw: 26
srwq:
  base: 124
  dep_per_unit: 1
state_root:
  base: 304
  dep_per_unit: 0
sub: 1
subi: 1
sw: 1
sww: 933
time: 1
tr: 58482
tro: 35540
wdam: 7
wdcm: 1
wddv: 3
wdmd: 8
wdml: 2
wdmm: 8
wdop: 2
wqam: 8
wqcm: 2
wqdv: 4
wqmd: 13
wqml: 2
wqmm: 8
wqop: 2
xor: 1
xori: 1
```

**Review by xgreenx — APPROVED** [review]

File: Cargo.lock [hunk 1]
```diff
@@ -1776,7 +1776,7 @@ version = "0.8.0"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "bef552e6f588e446098f6ba40d89ac146c8c7b64aade83c051ee00bb5d2bc18d"
 dependencies = [
- "uuid 1.5.0",
```
Comment (xgreenx) [inline Cargo.lock:1779]: It would be nice if you revert this file=) But it is optional

Comment (Dentosal) [reply Cargo.lock]: f4203c8b

**Review by xgreenx — APPROVED** [review]

**Pull request merged**